test/unit/org/apache/cassandra/Util.java

```diff
@@ -235,9 +235,6 @@ public static void createInitialRing(StorageService ss, IPartitioner partitioner
             ss.onChange(ep,
                         ApplicationState.STATUS_WITH_PORT,
                         new VersionedValue.VersionedValueFactory(partitioner).normal(Collections.singleton(endpointTokens.get(i))));
-            ss.onChange(ep,
```

Comment:
Wouldn't this be needed in mixed version clusters?

Reply:
My read of it was that we're just telling SS the endpoint is normal twice. That ApplicationState just gets us into the right branch [here](https://github.com/apache/cassandra/blob/5ed5e84613ef0e9664a774493db7d2604e3596e0/src/java/org/apache/cassandra/service/StorageService.java#L2262-L2278). I just noticed this while looking at how the status is used. It seemed superfluous to me, but if you don't want to change it, I think it's harmless to leave there.

Reply:
Mmmm icwym. Scratch that I was thinking along the lines whether we'd need send a STATUS in upgrades but this is not what we're doing here.